src/proteus/dswx_hls.py

```diff
@@ -996,8 +996,21 @@ def create_landcover_mask(copernicus_landcover_file,
     _update_landcover_array(hierarchy_combined, tree_aggregate_sum,
                             threshold_list[0], evergreen_forest_class)
 
+    # extract the year of the WorldCover dataset
+    worldcover_gdal_ds = gdal.Open(worldcover_file, gdal.GA_ReadOnly)
+    worldcover_metadata = worldcover_gdal_ds.GetMetadata()
+    del worldcover_gdal_ds
+    worldcover_datetime_format = '%Y-%m-%dT%H:%M:%SZ'
+    worldcover_time_start = datetime.strptime(worldcover_metadata['time_start'],
+                                              worldcover_datetime_format)
+    worldcover_time_end = datetime.strptime(worldcover_metadata['time_end'],
+                                            worldcover_datetime_format)
+
+    # the Worldcover dataset year is extracted from the average date times
+    worldcover_time_range = (worldcover_time_end - worldcover_time_start)
+    year = (worldcover_time_start + worldcover_time_range / 2.0).year - 2000
```

Comment:
Looking at the WorldCover user guides ([2020](https://esa-worldcover.s3.eu-central-1.amazonaws.com/v100/2020/docs/WorldCover_PUM_V1.0.pdf), [2021](https://esa-worldcover.s3.eu-central-1.amazonaws.com/v200/2021/docs/WorldCover_PUM_V2.0.pdf)), it looks like each WorldCover map only contains data from a single year. What scenario are you thinking of where the data would be from multiple years?

Otherwise, let's simplify this code.

Reply:
This is an example that I have here. From the WorldCover metadata:
```
 time_end=2020-12-31T23:59:59Z
 time_start=2020-01-01T00:00:00Z
```
I'm just taking the center (average) date because these two dates are in the "edge" of the year. The code will be robust to any changes that they apply on their side.

Reply:
Got it. Why not take the earlier of the two years, instead of the average? Per the DSWx-HLS product spec, the year in the LAND layer denotes "...first year development is present".

Reply:
For example, if they changed it to:
``` 
 time_end=2021-01-01T00:00:00Z
 time_start=2020-01-01T00:00:00Z
```
or to
``` 
 time_end=2020-12-31T23:59:59Z
 time_start=2019-12-31T23:59:59Z
```
I know these changes are unlikely though

Reply:
The average makes sense to me. It covers the normal cases like the existing one as well unlikely cases.

Reply:
This code refers to the WorldCover dataset that was given to the SAS. We want to know if this WorldCover dataset refers to year 2020 or 2021. For that, taking the center date is more robust.
The product specs refers to something else, it refers to the updating the LAND layer incrementally.

Reply:
John's initial idea was to have a "semi-static" global LAND layer. So, for example, OPERA would generate the global LAND layer for 2022. Then, for 2023, it would take the LAND from 2022 and update with the Copernicus Landcover 2023 and ESA Worldcover 2023 and generate the OPERA LAND 2023. If a pixel in the LAND layer 2022 was marked as urban/builtup 2022, and it's still urban/builtup in 2023, we'd keep the first time that it was marked as urban/builtup (in this example, 2022).

Reply:
I actually asked John today if we should keep with that classification considering that we don't have this "semi-static" global LAND layer anymore and we do write in the DSWx-HLS a description of the land cover inputs. He said he thinks it's useful to users to have the current urban/built up date (even though we are not keeping historical values anymore). So that date will always point to the WorldCover dataset that was provided to the SAS